src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/ReadyToRunMethod.cs

```diff
@@ -472,6 +713,14 @@ private void EnsureInitialized()
                     _gcInfo = new Amd64.GcInfo(_readyToRunReader.Image, gcInfoOffset, _readyToRunReader.Machine, _readyToRunReader.ReadyToRunHeader.MajorVersion);
                 }
             }
+            if (_pgoInfo == null)
+            {
+                _pgoInfo = _readyToRunReader.GetPgoInfoByKey(PgoInfoKey.FromReadyToRunMethod(this));
+                if (_pgoInfo == null)
+                {
+                    _pgoInfo = new object();
```

Comment:
Why do we init to `new object()` if we have no pgoInfo for a key?

Reply:
Mmm... my thought was to avoid a bunch of manual null checks, but I'll have to take a look and see if I was thinking straight at the time.